storage/sealing/garbage.go

```diff
@@ -6,25 +6,69 @@ import (
 	"io"
 	"math"
 	"math/rand"
+	"runtime"
+	"sync"
 
 	sectorbuilder "github.com/filecoin-project/go-sectorbuilder"
+	"github.com/hashicorp/go-multierror"
 	"golang.org/x/xerrors"
 
 	"github.com/filecoin-project/lotus/chain/actors"
 	"github.com/filecoin-project/lotus/chain/types"
 )
 
+func (m *Sealing) fastPledgeCommitment(size uint64, parts uint64) (commP [sectorbuilder.CommLen]byte, err error) {
+	piece := sectorbuilder.UserBytesForSectorSize((size / 127 + size) / parts)
```

Comment:
need to make sure that parts is a power of two, otherwise the path won't work out